To: vim_dev@googlegroups.com
Subject: Patch 7.3.063
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.063
Problem:    Win32: Running a filter command makes Vim lose focus.
Solution:   Use SW_SHOWMINNOACTIVE instead of SW_SHOWMINIMIZED. (Hong Xu)
Files:	    src/os_win32.c


*** ../vim-7.3.062/src/os_win32.c	2010-11-10 15:37:00.000000000 +0100
--- src/os_win32.c	2010-11-24 12:31:46.000000000 +0100
***************
*** 3185,3193 ****
       * It's nicer to run a filter command in a minimized window, but in
       * Windows 95 this makes the command MUCH slower.  We can't do it under
       * Win32s either as it stops the synchronous spawn workaround working.
       */
      if ((options & SHELL_DOOUT) && !mch_windows95() && !gui_is_win32s())
! 	si.wShowWindow = SW_SHOWMINIMIZED;
      else
  	si.wShowWindow = SW_SHOWNORMAL;
      si.cbReserved2 = 0;
--- 3185,3194 ----
       * It's nicer to run a filter command in a minimized window, but in
       * Windows 95 this makes the command MUCH slower.  We can't do it under
       * Win32s either as it stops the synchronous spawn workaround working.
+      * Don't activate the window to keep focus on Vim.
       */
      if ((options & SHELL_DOOUT) && !mch_windows95() && !gui_is_win32s())
! 	si.wShowWindow = SW_SHOWMINNOACTIVE;
      else
  	si.wShowWindow = SW_SHOWNORMAL;
      si.cbReserved2 = 0;
*** ../vim-7.3.062/src/version.c	2010-11-16 19:25:56.000000000 +0100
--- src/version.c	2010-11-24 12:32:52.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     63,
  /**/

-- 
SOLDIER: Where did you get the coconuts?
ARTHUR:  Through ... We found them.
SOLDIER: Found them?  In Mercea.  The coconut's tropical!
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
